### PR TITLE
feature/JM-6382&JM-6384: Staff-Management - View, Add, and Edit users 

### PR DESCRIPTION
**Pull request by callum-r-young**
### JIRA link (if applicable) ###
https://centralgovernmentcgi.atlassian.net/browse/JM-6382
https://centralgovernmentcgi.atlassian.net/browse/JM-6384

### Change description ###
Added system administration screens/flows for the following;
- Viewing all users (including searching, filtering, sorting, and pagination)
- Viewing a users record (differs depending on user type)
- Creating a new users
- Assigning courts to court users
- Editing existing users
Added court/bureau administration screens/flows for the following;
- Viewing users within their jurisdiction
- Viewing a users record (differs depending on user type)
- Editing existing users permissions
Added validation (and tests) for the required routes.
Added new auth middleware to check for manager role.

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: server/config/validation/create-users.spec.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    it('should try to validate an invalid request - invalid email entered', function() {
+      var mockRequest = {
+        name: 'Test Name',
+        email: 'testem,ail',
+      };
+
+      validatorResult = validate(mockRequest, validator.userDetails());
+
+      console.log(validatorResult);
```
Comment (SoheshJhuboo) [inline server/config/validation/create-users.spec.js:70]: Delete console log.

Comment (callum-r-young) [reply server/config/validation/create-users.spec.js]: @SoheshJhuboo Removed :)

**Comment by SoheshJhuboo** [comment]
@callum-r-young Code looks good. Pulled the PR and tested the features locally. Feature works well, however once a new user a created, I am unable to login as them.

**Comment by callum-r-young** [comment]
> @callum-r-young Code looks good. Pulled the PR and tested the features locally. Feature works well, however once a new user a created, I am unable to login as them.

@SoheshJhuboo I have spoken to @Ben-Edwards-cgi and he confirmed that new users will only be able to login once we transition over to the new azure login flow.

**Review by SoheshJhuboo — APPROVED** [review]

**Pull request merged**